### PR TITLE
Add threshold-entropy attack

### PR DESCRIPTION
**Pull request by lwsong**
Following our recent paper https://arxiv.org/pdf/2003.10595.pdf, I implement the code to run membership inference attacks by setting a threshold value on the prediction entropy value.
Compared to THRESHOLD_ATTACK, THRESHOLD_ENTROPY_ATTACK leverages the entropy value instead of the loss value.

### CODE REVIEW

File: tensorflow_privacy/privacy/membership_inference_attack/data_structures.py [hunk 1]
```diff
@@ -114,11 +114,12 @@ class AttackType(enum.Enum):
   RANDOM_FOREST = 'rf'
   K_NEAREST_NEIGHBORS = 'knn'
   THRESHOLD_ATTACK = 'threshold'
+  THRESHOLD_ENTROPY_ATTACK = 'threshold-entropy'
 
   @property
   def is_trained_attack(self):
     """Returns whether this type of attack requires training a model."""
-    return self != AttackType.THRESHOLD_ATTACK
+    return (self != AttackType.THRESHOLD_ATTACK) & (self != AttackType.THRESHOLD_ENTROPY_ATTACK)
```
Comment (CdavM) [inline tensorflow_privacy/privacy/membership_inference_attack/data_structures.py:122]: Can we use words instead, i.e. self != AttackType.THRESHOLD_ATTACK and self != AttackType.THRESHOLD_ENTROPY_ATTACK

File: tensorflow_privacy/privacy/membership_inference_attack/membership_inference_attack.py [hunk 2]
```diff
@@ -96,6 +96,19 @@ def _run_threshold_attack(attack_input: AttackInputData):
       attack_type=AttackType.THRESHOLD_ATTACK,
       roc_curve=roc_curve)
 
+def _run_threshold_entropy_attack(attack_input: AttackInputData):
```
Comment (CdavM) [inline tensorflow_privacy/privacy/membership_inference_attack/membership_inference_attack.py:99]: Can we add an end-to-end test for this to membership_inference_attack_test.py? Something like the "test_run_attack_threshold_calculates_correct_auc" test

**Review by CdavM — CHANGES_REQUESTED** [review]
Looks good, thanks Liwei! I left 2 comments

File: tensorflow_privacy/privacy/membership_inference_attack/data_structures.py (hunk 1 above)
Comment (lwsong) [reply tensorflow_privacy/privacy/membership_inference_attack/data_structures.py]: Fix it!

File: tensorflow_privacy/privacy/membership_inference_attack/membership_inference_attack.py (hunk 2 above)
Comment (lwsong) [reply tensorflow_privacy/privacy/membership_inference_attack/membership_inference_attack.py:99]: The test case is added!

**Comment by lwsong** [comment]
> Looks good, thanks Liwei! I left 2 comments

I have fixed these issues, please take another look, thanks.

**Review by CdavM — APPROVED** [review]
Looks good!
